### DISCUSSION TITLE
VPN mode in linux

### DISCUSSION THREAD

**Discussion by llaravell** [shaonhuang/V2rayX#10] [General]
Hi there.
Thanks for this awesome client, I have a question, does it support VPN mode?

**Comment by shaonhuang** [comment]
> Hi there. Thanks for this awesome client, I have a question, does it support VPN mode?

Hi buddy, thanks for asking. 
What do you mean by VPN mode? The logic of software is if you configure a server then choose the one you want to run. Turn on the service. It will run a proxy service in the background. The mode you switch is a cmd line(i.e. $gsettings set org.gnome.system.proxy mode manual) executed target to the Gnome environment in Linux. if you switch proxy mode to any mode between Pac or Global. The network traffic is using a proxy environment.

**Reply by shaonhuang** [reply]
I will probably add a custom execution cmd in setting pages for you to set if you want a different cmd to exec for adapting different systems.

**Reply by shaonhuang** [reply]
Currently, it is not supporting TUN mode because I am thinking of some issues and redesigns with the logger viewer page. If you mean the TUN mode(create a virtual network card and control all network flow goes under the network card) is the VPN mode.
